### PR TITLE
Optimizers added to flash

### PR DESCRIPTION
**Pull request by ananyahjha93**
## What does this PR do?

Adds LARS, LAMB, and linear warmup + cosine decay scheduler.

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/optimizers/lamb.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        >>> optimizer.step()
+
+    .. warning::
+        Since the default weight decay for LAMB is set to 0., we do not club together
+        0. weight decay and exclusion from layer adaptation like LARS. This would cause
+        the optimizer to exclude all layers from layer adaptation.
+    """
+
+    def __init__(
+        self,
```
Comment (Borda) [inline flash/core/optimizers/lamb.py:56]: pls add typing

File: flash/core/optimizers/lamb.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+                trust_ratio = 1.0
+                if not exclude_from_layer_adaptation:
+                    w_norm = torch.norm(p.data)
+                    g_norm = torch.norm(update)
+
+                    if w_norm > 0 and g_norm > 0:
+                        trust_ratio = w_norm / g_norm
+
+                p.add_(update, alpha=-group["lr"] * trust_ratio)
```
Comment (Borda) [inline flash/core/optimizers/lamb.py:160]: can we wrap this to some internal protected method?

**Review by Borda — COMMENTED** [review]
can you clarify that all is your/our own implementation?

File: flash/core/optimizers/lamb.py (hunk 2 above)
Comment (ananyahjha93) [reply flash/core/optimizers/lamb.py:161]: right now it works, I don't wanna change the code for an optimizer implementation which works, takes too much time to delve into it and make sure things do not get messed up.

**Review by tchaton — APPROVED** [review]
LGTM ! However, not sure Flash supports interval=step right now.

**Review by SeanNaren — APPROVED** [review]
After speaking with @ananyahjha93 the reason for not using a pre-existing implementation is out of correctness.

File: tests/core/test_scheduler.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    "lr, warmup_epochs, max_epochs, warmup_start_lr, eta_min",
+    [
+        (1, 10, 3200, 0.001, 0.0),
+        (1e-4, 40, 300, 1e-6, 1e-5),
+        (0.01, 1, 10, 0.0, 0.0),
+        (0.01, 0, 10, 0.0, 0.0),  # only cosine decay
+        (0.01, 10, 10, 0.0, 0.0),  # only linear warmup
+    ],
+)
+def test_LinearWarmupCosineAnnealingLR(tmpdir, lr, warmup_epochs, max_epochs, warmup_start_lr, eta_min):
```
Comment (ethanwharris) [inline tests/core/test_scheduler.py:33]: Can we move this file to `tests/core/optimizers/test_lr_shceduler.py` to mirror the code file structure? Also, this should be snake case: `test_linear_warmup_cosine_annealing_lr`

**Review by ethanwharris — COMMENTED** [review]
Looks good, can we have some tests for the LAMB and LARS optimizers?

File: tests/core/test_scheduler.py (hunk 3 above)
Comment (ananyahjha93) [reply tests/core/test_scheduler.py]: done

**Comment by ananyahjha93** [comment]
@ethanwharris how do you suggest writing tests for optimizers? Like you can check if a simple convex function loss function converges, but how do you check the correctness of lamb/lars's specifics? They are built on top of sgd/adam
